### PR TITLE
Use `vel.xy().length()` instead of `norm(vel.x,vel.y)`

### PR DESCRIPTION
**Pull request by hendjoshsr71**
This should be a non **functional** change.

I noticed the computation of `norm(vel.x,vel.y)` costs about 34--60 bytes each time it is used more than the equivalent `vel.xy().length()`

Multiply that by a bunch of times and you get back 1.2KB for Copter & 1KB for Plane and slightly less code.

### CODE REVIEW

File: ArduCopter/mode.cpp [hunk 1]
```diff
@@ -674,10 +674,10 @@ void Mode::land_run_horizontal_control()
         // there is any position estimate drift after touchdown. We
         // limit attitude to 7 degrees below this limit and linearly
         // interpolate for 1m above that
-        float attitude_limit_cd = linear_interpolate(700, copter.aparm.angle_max, get_alt_above_ground_cm(),
+        const float attitude_limit_cd = linear_interpolate(700, copter.aparm.angle_max, get_alt_above_ground_cm(),
```
Comment (peterbarker) [inline ArduCopter/mode.cpp:677]: Unrelated change.  But a good one anyway.

**Review by peterbarker — APPROVED** [review]
Nice find!

File: libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp [hunk 2]
```diff
@@ -1427,7 +1427,7 @@ void NavEKF3_core::alignMagStateDeclination()
 
     // rotate the NE values so that the declination matches the published value
     Vector3F initMagNED = stateStruct.earth_magfield;
-    ftype magLengthNE = norm(initMagNED.x,initMagNED.y);
+    ftype magLengthNE = initMagNED.xy().length();
```
Comment (rmackay9) [inline libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp:1430]: I guess there's no difference between norm and length() even for ftype?

**Comment by rmackay9** [comment]
I've reviewed the changes and it looks ok to me.  It would be very laborious to test each line that is changed so perhaps we just rely on inspection.

I've added one question above about making sure that length and norm work fine for ftype.

File: libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp (hunk 2 above)
Comment (hendjoshsr71) [reply libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp:1430]: I will plead a little ignorance here. I was surprised by the savings.
For this specific case initMagNED it is an fytpe so there is no difference. 

From what I can see in debugger and via inspection, T length(void) calls ftype norm(void) each function of  <template T> so there is output type narrowing that is possible if the inputs are floats and ftype -> is double?

**Comment by hendjoshsr71** [comment]
> I've reviewed the changes and it looks ok to me. It would be very laborious to test each line that is changed so perhaps we just rely on inspection.
> 
> I've added one question above about making sure that length and norm work fine for ftype.

This was 100% inspection and autotest only as it touches so many very specific test cases.

**Review by tridge — APPROVED** [review]
